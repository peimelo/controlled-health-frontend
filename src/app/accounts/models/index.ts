export * from './account-response.model';
export * from './account.model';
