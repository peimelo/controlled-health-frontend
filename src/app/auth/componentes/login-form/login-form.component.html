<mat-card>
  <mat-card-title>Login</mat-card-title>

  <mat-card-content>
    <form [formGroup]="form" (ngSubmit)="submit()">
      <p>
        <mat-form-field>
          <input
            formControlName="email"
            matInput
            placeholder="Email"
            required
            type="email"
          />
          <mat-error
            *ngFor="
              let error of formErrorService.mapErrors(form.get('email')?.errors)
            "
          >
            {{ error }}
          </mat-error>
        </mat-form-field>
      </p>

      <p>
        <mat-form-field>
          <input
            formControlName="password"
            matInput
            placeholder="Password"
            required
            type="password"
          />
          <mat-error
            *ngFor="
              let error of formErrorService.mapErrors(
                form.get('password')?.errors
              )
            "
          >
            {{ error }}
          </mat-error>
        </mat-form-field>
      </p>

      <p class="submitButton">
        <button
          color="primary"
          [disabled]="'login' | swIsLoading | async"
          mat-raised-button
          type="submit"
        >
          <fa-icon
            *ngIf="'login' | swIsLoading | async"
            [icon]="['fas', 'spinner']"
            [spin]="true"
          ></fa-icon>
          Login
        </button>
      </p>

      <p>
        <a mat-button routerLink="/forgot-password">Forgot password?</a>
      </p>

      <p>
        <a mat-button routerLink="/resend-confirmation">
          Resend confirmation email
        </a>
      </p>
    </form>
  </mat-card-content>
</mat-card>
