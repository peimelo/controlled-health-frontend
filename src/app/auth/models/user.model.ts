export interface Credentials {
  email: string;
  password: string;
}

export interface PasswordCombination {
  currentPassword?: string;
  password: string;
  passwordConfirmation: string;
}

export interface User {
  email: string;
  name: string;
}
