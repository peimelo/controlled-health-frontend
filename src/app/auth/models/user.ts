export interface Credentials {
  email: string;
  password: string;
}

export interface PasswordCombination {
  password: string;
  passwordConfirmation: string;
}

export interface User {
  email: string;
  name: string;
}
