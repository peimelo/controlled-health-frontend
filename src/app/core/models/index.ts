export * from './base-resource.model';
export * from './chart-result.model';
export * from './dialog-data.model';
export * from './height.model';
export * from './pagination.model';
export * from './serie.model';
export * from './unit.model';
export * from './weight.model';
