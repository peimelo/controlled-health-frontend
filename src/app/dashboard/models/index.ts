export * from './chart-result.model';
export * from './dashboard.model';
export * from './serie.model';
