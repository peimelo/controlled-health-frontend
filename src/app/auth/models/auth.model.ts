import { PasswordCombination, User } from './user.model';

export interface CreateAccountRequest extends PasswordCombination {
  email: string;
}

export interface MessageResponse {
  message: string;
}

export interface UserDataResponse {
  data: User;
}

export interface UpdateAccountResponse
  extends MessageResponse,
    UserDataResponse {}
