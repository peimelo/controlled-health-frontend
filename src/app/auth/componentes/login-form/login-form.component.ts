import { Component, EventEmitter, Input, Output } from '@angular/core';
import { FormBuilder, Validators } from '@angular/forms';
import { FormErrorService } from '../../../core/services/form-error.service';
import { Credentials } from '../../models';

@Component({
  selector: 'app-login-form',
  templateUrl: './login-form.component.html',
  styleUrls: ['./login-form.component.scss'],
})
export class LoginFormComponent {
  form = this.fb.group({
    email: ['', [Validators.email, Validators.required]],
    password: ['', Validators.required],
  });

  @Input()
  set pending(isPending: boolean) {
    if (isPending) {
      this.form.disable();
    } else {
      this.form.enable();
    }
  }

  @Output() private submit = new EventEmitter<Credentials>();

  constructor(
    private fb: FormBuilder,
    public readonly formErrorService: FormErrorService
  ) {}

  onSubmit(): void {
    if (this.form.valid) {
      this.submit.emit(this.form.value);
    }
  }
}
