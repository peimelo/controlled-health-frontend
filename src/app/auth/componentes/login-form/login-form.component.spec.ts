import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { Credentials } from '../../models';
import { LoginFormComponent } from './login-form.component';

describe('Login Page', () => {
  let fixture: ComponentFixture<LoginFormComponent>;
  let instance: LoginFormComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [ReactiveFormsModule],
      declarations: [LoginFormComponent],
      schemas: [NO_ERRORS_SCHEMA],
    });

    fixture = TestBed.createComponent(LoginFormComponent);
    instance = fixture.componentInstance;
  });

  it('should compile', () => {
    fixture.detectChanges();

    (<any>expect(fixture)).toMatchSnapshot();
  });

  it('should disable the form if pending', () => {
    instance.pending = true;

    fixture.detectChanges();

    (<any>expect(fixture)).toMatchSnapshot();
  });

  it('should emit an event if the form is valid when submitted', () => {
    const credentials: Credentials = {
      email: 'user@email.com',
      password: 'password',
    };
    instance.form.setValue(credentials);

    spyOn(instance.submitted, 'emit');
    instance.submit();

    expect(instance.submitted.emit).toHaveBeenCalledWith(credentials);
  });
});
