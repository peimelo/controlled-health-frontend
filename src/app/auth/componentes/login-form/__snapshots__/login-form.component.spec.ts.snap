// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should compile 1`] = `
<app-login-form
  fb={[Function FormBuilder]}
  form={[Function FormGroup]}
  formErrorService={[Function FormErrorService]}
  submitted={[Function EventEmitter_]}
>
  <mat-card>
    <mat-card-title>
      Login
    </mat-card-title>
    <mat-card-content>
      <form
        class="ng-untouched ng-pristine ng-invalid"
        novalidate=""
      >
        <p>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="email"
              matinput=""
              placeholder="Email"
              required=""
              type="email"
            />
            
            <mat-error>
               Field is required. 
            </mat-error>
          </mat-form-field>
        </p>
        <p>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="password"
              matinput=""
              placeholder="Password"
              required=""
              type="password"
            />
            
            <mat-error>
               Field is required. 
            </mat-error>
          </mat-form-field>
        </p>
        <p
          class="submitButton"
        >
          <button
            color="primary"
            mat-raised-button=""
            type="submit"
          >
             Login 
          </button>
        </p>
        <p>
          <a
            mat-button=""
            routerlink="/forgot-password"
          >
            Forgot password?
          </a>
        </p>
        <p>
          <a
            mat-button=""
            routerlink="/resend-confirmation"
          >
             Resend confirmation email 
          </a>
        </p>
      </form>
    </mat-card-content>
  </mat-card>
</app-login-form>
`;

exports[`Login Page should disable the form if pending 1`] = `
<app-login-form
  fb={[Function FormBuilder]}
  form={[Function FormGroup]}
  formErrorService={[Function FormErrorService]}
  submitted={[Function EventEmitter_]}
>
  <mat-card>
    <mat-card-title>
      Login
    </mat-card-title>
    <mat-card-content>
      <form
        class="ng-untouched ng-pristine"
        novalidate=""
      >
        <p>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine"
              disabled=""
              formcontrolname="email"
              matinput=""
              placeholder="Email"
              required=""
              type="email"
            />
            
          </mat-form-field>
        </p>
        <p>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine"
              disabled=""
              formcontrolname="password"
              matinput=""
              placeholder="Password"
              required=""
              type="password"
            />
            
          </mat-form-field>
        </p>
        <p
          class="submitButton"
        >
          <button
            color="primary"
            mat-raised-button=""
            type="submit"
          >
             Login 
          </button>
        </p>
        <p>
          <a
            mat-button=""
            routerlink="/forgot-password"
          >
            Forgot password?
          </a>
        </p>
        <p>
          <a
            mat-button=""
            routerlink="/resend-confirmation"
          >
             Resend confirmation email 
          </a>
        </p>
      </form>
    </mat-card-content>
  </mat-card>
</app-login-form>
`;
